docs/spec/baseapp/WIP_abci_application.md

```diff
@@ -0,0 +1,46 @@
+# ABCI application
+
+The `BaseApp` struct fulfills the tendermint-abci `Application` interface. 
```

Comment:
Imho, I still think this has a significant amount of overlap with the new baseapp docs found [here](https://github.com/cosmos/cosmos-sdk/blob/develop/docs/reference/baseapp.md).